vnext/Microsoft.ReactNative/XamlView.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "XamlView.h"
+#include <UI.Xaml.Controls.Primitives.h>
+#include <UI.Xaml.Documents.h>
+
+namespace react::uwp {
+
+xaml::XamlRoot TryGetXamlRoot(const XamlView &view) {
+  xaml::XamlRoot root{nullptr};
+  if (auto uielement = view.try_as<xaml::UIElement>()) {
+    root = uielement.XamlRoot();
+  } else if (auto textelement = view.try_as<xaml::Documents::TextElement>()) {
+    root = textelement.XamlRoot();
+  } else if (auto flyout = view.try_as<xaml::Controls::Primitives::FlyoutBase>()) {
+    root = flyout.XamlRoot();
+  }
+  return root;
+}
+
+comp::Compositor GetCompositor(const XamlView &view) {
+  if (auto window = xaml::Window::Current()) {
+    return window.Compositor();
+  }
+#ifdef USE_WINUI3
+  else {
+    return TryGetXamlRoot(view).Compositor();
```

Comment:
> return TryGetXamlRoot(view).Compositor(); [](start = 3, length = 42)

I suggest adding an assert that the view is loaded to help catch uses of the compositor before it's available.

Reply:
the XamlRoot will be null if it hasn't been added to the tree so we'll get a null AV there which will hopefully be clear enough :)